stl/inc/algorithm

```diff
@@ -10163,16 +10163,24 @@ _NODISCARD constexpr const _Ty& clamp(const _Ty& _Val, const _Ty& _Min_val, cons
         return _Val;
     }
 #endif // _ITERATOR_DEBUG_LEVEL == 2
+    if constexpr ((is_integral_v<_Ty> || is_enum_v<_Ty> || is_pointer_v<_Ty>
+#ifdef _M_FP_FAST
+                      || is_floating_point_v<_Ty>
+#endif //  _M_FP_FAST
+                      ) &&(is_same_v<_Remove_cvref_t<_Pr>, less<>> || is_same_v<_Remove_cvref_t<_Pr>, less<_Ty>>) ) {
+        // Avoid short-circuit evaluation to promote cmov generation; see GH-2334
+        return (_STD min) ((_STD max) (_Val, _Min_val), _Max_val);

```

Comment:
I'm not too concerned about problems here, but we don't appear to have comprehensive type coverage of `clamp` in our tests (I think this might have been a function that we primarily relied on libcxx for).

I'd like to see tests for all of the `_Ty` types that can activate this optimization. It would be nice to have each of the 3 possible results (clamped to min, unchanged, clamped to max). I don't think we need to cover `less<>` and `less<T>` separately (but wouldn't complain if we did).

Mostly this is because `min` and `max` have SAL annotations so I have vague worries about compiler bugs/warnings.

Reply:
Not using `min` / `max` anymore -- direct ternary would be better for debug codegen.
Still agree tests would be good, will add some.